olp-cpp-sdk-dataservice-read/tests/ApiTest.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <gmock/gmock-matchers.h>
+#include <matchers/NetworkUrlMatchers.h>
+#include <mocks/NetworkMock.h>
+
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/client/OlpClientFactory.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+
+#include "generated/api/QueryApi.h"
+
+namespace {
+constexpr auto kLayerid = "test-layer";
+constexpr auto kQuadKey = "23618401";
+constexpr auto kNodeBaseUrl = "https://some.node.base.url/stream/v2/catalogs/"
+                              "hrn:here:data::olp-here-test:hereos-internal-test-v2";
+constexpr auto kUrlQuadTreeIndexVolatile =
+    R"(https://some.node.base.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2/layers/test-layer/quadkeys/23618401/depths/2)";
+
+constexpr auto kHttpResponseEmpty = R"jsonString()jsonString";
+constexpr auto kHttpResponseQuadTreeIndexVolatile = R"jsonString({ "parentQuads": [ { "additionalMetadata": "string", "checksum": "string", "compressedDataSize": 0, "dataHandle": "675911FF6236B7C7604BF8B105F1BB58", "dataSize": 0, "crc": "c3f276d7", "partition": "73982", "version": 0 } ], "subQuads": [ { "additionalMetadata": "string", "checksum": "291f66029c232400e3403cd6e9cfd36e", "compressedDataSize": 200, "dataHandle": "1b2ca68f-d4a0-4379-8120-cd025640510c", "dataSize": 1024, "crc": "c3f276d7", "subQuadKey": "string", "version": 1 } ] })jsonString";
+
+using testing::_;
+using namespace ::testing;
+using namespace olp;
+using namespace olp::client;
+using namespace olp::dataservice::read;
+using namespace olp::tests::common;
+
+class ApiTest : public Test {
+ protected:
+   void SetUp() override {
+    network_mock_ = std::make_shared<NetworkMock>();
+
+    settings_ = std::make_shared<OlpClientSettings>();
+    settings_->network_request_handler = network_mock_;
+
+    client_ = OlpClientFactory::Create(*settings_);
+    client_->SetBaseUrl(kNodeBaseUrl);
+  }
+   void TearDown() override { network_mock_.reset(); }
+
+ protected:
+  std::shared_ptr<OlpClientSettings> settings_;
+  std::shared_ptr<OlpClient> client_;
+  std::shared_ptr<NetworkMock> network_mock_;
+};
+
+TEST_F(ApiTest, QuadTreeIndexVolatile) {
```

Comment:
This is not enough. Where are the negative tests? E.g. invalid OlpClient, invalid layer_id, etc. You need to make sure that you do not crash in case users provide garbage.